### PR TITLE
Sam/1341

### PR DESCRIPTION
**Pull request by bleakcon**
Added regex for validating that a username:

- does not begin with a non-alphanumeric character
- does not end with a non-alphanumeric character
- in addition to the above constraints cannot contain any non-alphanumeric characters other than - _ .

### CODE REVIEW

File: src/utils/isValidUsername.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+const isValidUsername = (username: string): boolean => {
+  const result: string[] | null = username.match(/^[a-z0-9][a-z0-9-_.]*[a-z0-9]$/i)
```
Comment (MihaiMadeT) [inline src/utils/isValidUsername.ts:2]: what about upper case letter?

File: test/utils/validateUserProperties.test.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+      const result = isValidUsername(proposedUsername)
+      // Then
+      expect(result).toBe(false)
+    })
+  })
+
+  describe("AND a space is at the end of the proposed username", () => {
+    it("SHOULD return false", () => {
+      // Given
+      const proposedUsername = " user name"
```
Comment (MihaiMadeT) [inline test/utils/validateUserProperties.test.ts:61]: the space is not at the end of the username

**Review by MihaiMadeT — COMMENTED** [review]
One or 2 comments, but otherwise looks good

File: test/utils/validateUserProperties.test.ts (hunk 2 above)
Comment (bleakcon) [reply test/utils/validateUserProperties.test.ts:61]: whoops -_-

File: src/utils/isValidUsername.ts (hunk 1 above)
Comment (bleakcon) [reply src/utils/isValidUsername.ts:2]: the i at the end allows for upper case

File: test/utils/validateUserProperties.test.ts (hunk 2 above)
Comment (bleakcon) [reply test/utils/validateUserProperties.test.ts]: Review comment 1

**Review by MihaiMadeT — APPROVED** [review]

**Pull request merged**